### PR TITLE
Add support for listing read/only and read/write images

### PR DESCRIPTION
**Pull request by rhatdan**
When removing --all images prune images only attempt to remove read/write images,
ignore read/only images

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: cmd/podman/cliconfig/config.go [hunk 1]
```diff
@@ -57,6 +57,8 @@ type ImagesValues struct {
 	Noheading bool
 	NoTrunc   bool
 	Quiet     bool
+	RO        bool
```
Comment (mheon) [inline cmd/podman/cliconfig/config.go]: Can we do this as a filter instead? Keeps the frontend cleaner. `--filter readonly=true/false` or similar?

Comment (rhatdan) [reply cmd/podman/cliconfig/config.go]: Much better idea, thanks.

**Comment by giuseppe** [comment]
LGTM

File: docs/podman-images.1.md [hunk 2]
```diff
@@ -23,6 +23,26 @@ Show image digests
...
+  Filters:
+
+  **after**
+    Filter on images created after the given time.Time
+
+  **before**
+    Filter on images created before the given time.Time
+
+  **dangling=true**
+    Show dangling images.  Dangling images are layers that have no relationship to any tagged images. They do not serve a purpose and consume disk space.
```
Comment (TomSweeneyRedHat) [inline docs/podman-images.1.md]: Suggest:
"Show dangling images.   Dangling images are a file system layer that was used in a previous build of an image and are no longer referenced by any active images.  They are denoted with the `<none>` tag, consume disk  space and server no active purpose."

File: docs/podman-images.1.md [hunk 3]
```diff
@@ -23,6 +23,26 @@ Show image digests
...
+
+  **label**
+    Filter by images labels key and/or value
```
Comment (TomSweeneyRedHat) [inline docs/podman-images.1.md]: missing period at end

File: docs/podman-images.1.md (hunk 2 above)
Comment (TomSweeneyRedHat) [reply docs/podman-images.1.md]: just updated my previous comment to fix some pre-tea grammar

Comment (rhatdan) [reply docs/podman-images.1.md]: Nice job, I like your description.

File: docs/podman-images.1.md (hunk 3 above)
Comment (rhatdan) [reply docs/podman-images.1.md]: Fixed.

**Comment by rhatdan** [comment]
No command completions, we don't handle this sub-options currently.

Added test.

File: pkg/varlinkapi/images.go [hunk 4]
```diff
@@ -98,6 +99,8 @@ func (i *LibpodAPI) GetImage(call iopodman.VarlinkCall, id string) error {
 		return err
 	}
 
+	fmt.Println("DAN isReadOnly %d", newImage.IsReadOnly())
```
Comment (haircommander) [inline pkg/varlinkapi/images.go]: a left over debug?

Comment (vrothberg) [reply pkg/varlinkapi/images.go]: Sometimes we just cannot write to DAN.

Comment (rhatdan) [reply pkg/varlinkapi/images.go]: Will open a PR to fix.

**Pull request merged**